test/elm/datetime/test.coffee

```diff
@@ -154,33 +156,78 @@ describe 'Time', ->
 describe 'Today', ->
   @beforeEach ->
     setup @, data
+    @ctx = new PatientContext(@ctx.library, @ctx.patient, @ctx.codeService, @ctx.parameters)
 
   it 'should return only day components and timezone of today', ->
-    jsDate = new Date()
     today = @todayVar.exec @ctx
     today.isTime().should.be.false()
-    today.year.should.equal jsDate.getFullYear()
-    today.month.should.equal jsDate.getMonth() + 1
-    today.day.should.equal jsDate.getDate()
-    today.timezoneOffset.should.equal jsDate.getTimezoneOffset() / 60 * -1
+    today.year.should.equal @ctx.getExecutionDateTime().year
+    today.month.should.equal @ctx.getExecutionDateTime().month
+    today.day.should.equal @ctx.getExecutionDateTime().day
+    today.timezoneOffset.should.equal @ctx.getTimezoneOffset()
+    should.not.exist(today[field]) for field in [ 'hour', 'minute', 'second', 'millisecond' ]
+
+  it 'should return only day components and timezone of today using the passed in timezone', ->
+    @ctx = new PatientContext(@ctx.library, @ctx.patient, @ctx.codeService, @ctx.parameters, DT.DateTime.fromDate(new Date(), '0'))
+    today = @todayVar.exec @ctx
+    today.isTime().should.be.false()
+    today.year.should.equal @ctx.getExecutionDateTime().year
+    today.month.should.equal @ctx.getExecutionDateTime().month
+    today.day.should.equal @ctx.getExecutionDateTime().day
+    today.timezoneOffset.should.equal "0"
     should.not.exist(today[field]) for field in [ 'hour', 'minute', 'second', 'millisecond' ]
 
+  it 'should throw an exception because no execution datetime has been set', ->
+    try
+      @ctx = new PatientContext(@ctx.library, @ctx.patient, @ctx.codeService, @ctx.parameters, DT.DateTime.fromDate(new Date(), '0'))
+      @ctx.executionDateTime = @ctx.executionDateTime = null
+      @todayVar.exec(@ctx).should.equal "No Execution DateTime has been set"
+    catch
```

Comment:
Unless I'm misunderstanding it, I don't think this test actually does what it says it does.

Reply:
On second read, it _does_ test it, but it's a little confusing as written.  If it didn't throw, it would fail because `@todayVar` can never equal "No Execution DateTime has been set" -- but since it does throw it goes to the empty catch.

At the least, I'd prefer to see a comment like:
```
// the following line should never be executed because the line above is expected to throw
```

But ideally, line 184 would be something more like:
```
should(() => @todayVar.exec(@ctx)).throw(/No Execution DateTime has been set/)
```